main.tf

```diff
@@ -59,24 +100,72 @@ data "aws_iam_policy_document" "assume_role_policy" {
 
 data "aws_iam_policy_document" "allows_s3" {
   statement {
-    sid    = "S3AccessForAWSTransferusers"
+    sid    = "AllowListingOfUserFolder"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket"
+    ]
+
+    resources = [
+      data.aws_s3_bucket.landing.arn
+    ]
+  }
+
+  statement {
+    sid    = "HomeDirObjectAccess"
     effect = "Allow"
 
-    actions = ["s3:*"]
+    actions = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion",
+      "s3:GetObjectVersion",
+      "s3:GetObjectACL",
+      "s3:PutObjectACL"
+    ]
 
     resources = [
-      "arn:aws:s3:::*"
+      "${data.aws_s3_bucket.landing.arn}/*"
     ]
   }
 }
 
+data "aws_iam_policy_document" "logging" {
+  statement {
+    sid    = "CloudWatchAccessForAWSTransfer"
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents"
+    ]
+
+    resources = ["*"]
+  }
+}
+
 resource "aws_iam_policy" "default" {
   name   = module.iam_label.id
   policy = data.aws_iam_policy_document.allows_s3.json
 }
 
+resource "aws_iam_policy" "logging" {
+  name   = module.logging_label.id
```

Comment:
count logic should be on all resources and data sources. our modules already have it covered with the `context`.